### ISSUE TITLE
Error when dataframe contains a column named `index`

### ISSUE THREAD

**Issue by yarontaubi** [bloomberg/ipydatagrid#250]
This code will work fine:
```
df = pd.DataFrame({'a': [1]*3, 'b':['a']*3})
DataGrid(df)
```

this will fail on `ValueError: The dataframe must not contain duplicate column names.`:
```
df = pd.DataFrame({'a': [1]*3, 'index':['a']*3})
DataGrid(df)
```

This error, even if unavoidable, is not very informative.

**Comment by ibdafna** [comment]
Hello 👋 @yarontaubi and welcome to the `ipydatagrid` community!

Thank you for reporting this issue - should definitely not happen. I'll take a look!

**Comment by ibdafna** [comment]
@kaiayoung this error seems to stem from the way `pandas` generates the JSON Table Schema. This is the output for the following DataFrame:

```python
df = pd.DataFrame(data={'index':[1,2,3], 'col2':[4,5,6]})
```

```python
{
    'fields': [
        {'name': 'index', 'type': 'integer'}, 
        {'name': 'index', 'type': 'integer'}, 
        {'name': 'col2', 'type': 'integer'}, 
    ], 
    'primaryKey': [
        'index'
    ], 
    'pandas_version': '0.20.0'
}
```

The call blindly adds the as `primaryKey` without checking for duplicates, which ends up in two column identical column names. This is okay in `pandas`-land, but not in JSON Table Schema . One way to solve this is to simply disallow any DataFrames with "index" as a column name. Another way is to detect is a passed DataFrame has a column named "index", and if it does, change the `primaryKey` pandas generates such that it is not named "index". This means that the index column in ipydatagrid will have a different name such as "id" or something similar:

```python
{
    'fields': [
        {'name': 'id', 'type': 'integer'}, 
        {'name': 'index', 'type': 'integer'}, 
        {'name': 'col2', 'type': 'integer'}, 
    ], 
    'primaryKey': [
        'id'
    ], 
    'pandas_version': '0.20.0'
}
```

I'm more inclined to go with the latter. What are your thoughts?

**Comment by ibdafna** [comment]
Resolved via #260. I will cut a release by the end of next week!
